

# Add or remove the repetitive read True = Add, False = remove
bool add

# Device to read from
uint8 MSP430=0
uint8 IMU=1
uint8 device

# Address of register to read from
uint8 addr

# Number of bytes to read
uint8 size

# Topic to publish to after a read
string topic

----

# Success
bool success

# Failure code
uint8 OK = 0
# Remove failures
uint8 NO_ID_MATCH = 1
uint8 UNKOWN_REMOVE_FAILURE = 2
# Add failures
uint8 ID_ALREADY_EXIST = 3
uint8 REPETITIVE_READ_AT_CAPACITY = 4
uint8 UNKOWN_ADD_FAILURE = 5
uint8 failure_code
